modules/solid_mechanics/src/materials/CombinedPlasticityStressUpdate.C

```diff
@@ -0,0 +1,258 @@
+    _isotropic_hardening_slope(0.0),
+    _plastic_strain(this->template declareGenericProperty<RankTwoTensor, is_ad>(
+        _base_name + _plastic_prepend + "plastic_strain")),
+    _plastic_strain_old(this->template getMaterialPropertyOld<RankTwoTensor>(
+        _base_name + _plastic_prepend + "plastic_strain")),
+    _backstress(this->template declareGenericProperty<RankTwoTensor, is_ad>("backstress")),
+    _backstress_old(this->template getMaterialPropertyOld<RankTwoTensor>("backstress")),
+    _kinematic_hardening_modulus(this->template getParam<Real>("kinematic_hardening_modulus")),
+    _gamma(this->template getParam<Real>("material_constant_gamma")),
+    Q(this->template getParam<Real>("Q")),
+    b(this->template getParam<Real>("b")),
+    _isotropic_hardening_variable(this->template declareGenericProperty<Real, is_ad>(
+        _base_name + "isotropic_hardening_variable")),
+    _isotropic_hardening_variable_old(
+        this->template getMaterialPropertyOld<Real>(_base_name + "isotropic_hardening_variable")),
+    _kinematic_hardening_variable(
+        this->template declareGenericProperty<Real, is_ad>("kinematic_hardening_variable")),
+    _kinematic_hardening_variable_old(
+        this->template getMaterialPropertyOld<Real>("kinematic_hardening_variable")),
+    _temperature(this->template coupledGenericValue<is_ad>("temperature"))
+{
+  if (parameters.isParamSetByUser("yield_stress") && _yield_stress <= 0.0)
+    mooseError("Yield stress must be greater than zero");
+  // Both of these parameters are given default values by derived classes, which makes them valid
+  if (_yield_stress_function == nullptr && !this->isParamValid("yield_stress"))
+    mooseError("Either yield_stress or yield_stress_function must be given");
+  if (!parameters.isParamValid("isotropic_hardening_constant") &&
+      !this->isParamValid("isotropic_hardening_function"))
+    mooseError(
+        "Either isotropic_hardening_constant or isotropic_hardening_function must be defined");
+  if (parameters.isParamSetByUser("isotropic_hardening_constant") &&
+      this->isParamValid("isotropic_hardening_function"))
+    mooseError(
+        "Only the isotropic_hardening_constant or only the isotropic_hardening_function can be "
+        "defined but not both");
+}
+
+template <bool is_ad>
+void
+CombinedPlasticityStressUpdateTempl<is_ad>::initQpStatefulProperties()
+{
+  _isotropic_hardening_variable[_qp] = 0.0;
+  _kinematic_hardening_variable[_qp] = 0.0;
+  _plastic_strain[_qp].zero();
+  _backstress[_qp].zero();
+}
+
+template <bool is_ad>
+void
+CombinedPlasticityStressUpdateTempl<is_ad>::propagateQpStatefulProperties()
+{
+  _isotropic_hardening_variable[_qp] = _isotropic_hardening_variable_old[_qp];
+  _kinematic_hardening_variable[_qp] = _kinematic_hardening_variable_old[_qp];
+  _plastic_strain[_qp] = _plastic_strain_old[_qp];
+  _backstress[_qp] = _backstress_old[_qp];
```

Comment:
move `_backstress[_qp] = _backstress_old[_qp];` to backStressRadialReturnStressUpdate